Makefile.am

```diff
@@ -120,8 +120,8 @@ libkmod_libkmod_internal_la_LDFLAGS = $(AM_LDFLAGS) \
 libkmod_libkmod_internal_la_DEPENDENCIES  = $(libkmod_libkmod_la_DEPENDENCIES)
 libkmod_libkmod_internal_la_LIBADD = $(libkmod_libkmod_la_LIBADD)
 
-pkgconfig_DATA = libkmod/libkmod.pc
-noarch_pkgconfig_DATA = tools/kmod.pc
+pkgconfigdir = $(libdir)/pkgconfig
```

Comment:
Please do NOT hard-code this location. Use the PKG_INSTALLDIR macro in configure.ac to make it configurable.

Reply:
That sounds like it might be the right thing to do. If you don't care about multiple architectures you can set it do $(datadir)/pkgconfig

Reply:
I think you could leave my commit as-is and pass `--with-noarch-pkgconfigdir=...` to configure to accomplish your goal.

That way by default libkmod.pc ends up in libdir and kmod.pc ends up in datadir, which makes more sense for a typical system installation of kmod.

Reply:
That also has the advantage of making the install locations of libkmod.pc and kmod.pc independent of each other. If we use pkgconfig_DATA for both files we would lose the ability to install them in separate locations.

Reply:
> That also has the advantage of making the install locations of libkmod.pc and kmod.pc independent

Looks like that's the core of the misunderstanding.

They are not independent.

What you can do on a binary distribution is to package the library and the binary separately, and swap between binaries for different architectures.

Or use binaries with prefix/suffix. And then match the pkgconfig path. Unfortunately, the proposal to dump the data by the tool itself was rejected, and that makes the life of people who really do need multiple different kmod builds more complicated.

Reply:
Ok, I guess I can live with kmod.pc in pkgconfigdir since it does contain data relevant to libkmod.

I still think using PKG_INSTALLDIR in configure.ac is a nice improvement though.

Reply:
It might have been nice to separate these concerns more cleanly when initially implementing this.

depmod can override the module directory, the other data could be put in libkmod.pc instead.

libkmod users that do not specify the module directory get a fallback that is not specified in libkmod.pc.

Reply:
The patch to make use of this functionality in the kernel was not accepted so far so I suppose the way the information is conveyed can still be changed.

The objection to the chosen approach was that kernel build would require the `jq` tool for the kernel build itself (as opposed to additonal tools build) whereas pkgconfig is already required.

However, it turns out that pkgconfig is not a great match for this purpose.

Reply:
besides needing jq you´d also need depmod for the target, which was never required since depmod always worked for foreign archs

Reply:
it's needed only if the module directory differs per architecture or whatever other variable.

Reply:
Rmoved kmod.pc, added additional depmod parameter to print the module directory instead.

Reply:
No, as I said this is a regression on what we provided.  Developers shouldn't need to run depmod from a foreign architecture. depmod always worked in that case and I don't see a need for that.

Reply:
Removing the pkgconfig would also regress for every package checking for kmod.pc. Let's move the pkgconfig discussion to a different PR as it's orthogonal to the rest of the changes here.

Reply:
A regression in what sense, what was provided, etc?

We need some way to convey what the default is for the depmod the user is using for building their kernel, native, cross, some random local build or otherwise.

That's what the new --moduledir-default option provides, and the pkgconfig file provided in a more awkward way.

Previously the path was hardcoded, so there was no point in having a 'foreign' (cross-architecture or cross-distribution or custom built) kmod, it would do the same.

Now it can do differently. An option to override one thing (module directory) is provided, other things cannot be overridden (eg. the config file locations).

It may now make sense to run a 'foreign' kmod to get the correct defaults for the target system automatically, without the need to ask an oracle what the correct value is, and passing it to kernel makefile.

It's awkward trying to read the default value from a file that is searched on a different path than the tool itself, these paths need to be matched if there is a possibility that multiple versions of the tool exist, for whatever reason.